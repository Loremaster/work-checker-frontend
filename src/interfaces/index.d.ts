export interface IStore {
  token: string | null;
  error: Error | null;
  user: {
    role: string | null;
  };

  users?: IUserList[];
}

export interface IUserList {
  id: number;
  role: string;
  email: string;
}
