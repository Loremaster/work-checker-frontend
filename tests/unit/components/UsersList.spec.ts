import { mount, createLocalVue } from "@vue/test-utils";
import { IUserList } from "@/interfaces";
import { insertDataAppDiv } from "../helpers/vuetify-helper";
import Vuetify from "vuetify";
import Vue from "vue";
import Vuex from "vuex";
import Component from "@/components/UsersList.vue";

const localVue = createLocalVue();

localVue.use(Vuex);

Vue.use(Vuetify);

let getters: any;
let actions: any;
let store: any;
let wrapper: any;

describe("UsersList.vue", () => {
  beforeEach(() => {
    actions = { fetchUsers: jest.fn() };
    getters = { users: () => [] };
  });

  const setup = () => {
    insertDataAppDiv();
    store = new Vuex.Store({ actions, getters });
    wrapper = mount(Component, { store, localVue });
  };

  it("renders correctly", () => {
    setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("fetches users", () => {
    setup();

    expect(actions.fetchUsers).toHaveBeenCalledWith(expect.anything(), undefined, undefined);
  });

  it("renders users", () => {
    const users: IUserList[] = [{ id: 1, email: "some@user.com", role: "admin" }];
    getters = { users: () => users };
    setup();

    expect(wrapper.text()).toContain("some@user.com");
    expect(wrapper.text()).toContain("admin");
  });
});
