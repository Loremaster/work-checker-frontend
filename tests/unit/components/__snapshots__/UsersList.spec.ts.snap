// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UsersList.vue renders correctly 1`] = `
<div class="elevation-1">
  <div class="v-table__overflow">
    <table class="v-datatable v-table theme--light">
      <thead>
        <tr>
          <th role="columnheader" scope="col" aria-label="Email: Sorted ascending. Activate to sort descending." aria-sort="ascending" tabindex="0" class="column sortable active asc text-xs-left">Email<i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">arrow_upward</i></th>
          <th role="columnheader" scope="col" aria-label="Role: Not sorted. Activate to sort ascending." aria-sort="none" tabindex="0" class="column sortable text-xs-left">Role<i aria-hidden="true" class="v-icon material-icons theme--light" style="font-size: 16px;">arrow_upward</i></th>
        </tr>
        <tr class="v-datatable__progress">
          <th colspan="2" class="column"></th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td colspan="2" class="text-xs-center">No data available</td>
        </tr>
      </tbody>
    </table>
  </div>
  <div class="v-datatable v-table theme--light">
    <div class="v-datatable__actions">
      <div class="v-datatable__actions__select">Rows per page:<div role="combobox" class="v-input v-text-field v-select v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light">
          <div class="v-input__control">
            <div class="v-input__slot">
              <div class="v-select__slot">
                <div class="v-select__selections">
                  <div class="v-select__selection v-select__selection--comma">5</div><input aria-label="Rows per page:" readonly="readonly" type="text" autocomplete="on" aria-readonly="false">
                </div>
                <div class="v-input__append-inner">
                  <div class="v-input__icon v-input__icon--append"><i aria-hidden="true" class="v-icon material-icons theme--light">arrow_drop_down</i></div>
                </div>
              </div>
              <div class="v-menu"></div>
            </div>
          </div>
        </div>
      </div>
      <div class="v-datatable__actions__range-controls">
        <div class="v-datatable__actions__pagination">â€“</div><button disabled="disabled" type="button" class="v-btn v-btn--disabled v-btn--flat v-btn--icon theme--light" aria-label="Previous page">
          <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light">chevron_left</i></div>
        </button><button disabled="disabled" type="button" class="v-btn v-btn--disabled v-btn--flat v-btn--icon theme--light" aria-label="Next page">
          <div class="v-btn__content"><i aria-hidden="true" class="v-icon material-icons theme--light">chevron_right</i></div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
